### PR TITLE
base: aktualizr: Split aktualizr-lite apps packages

### PR DESCRIPTION
**Pull request by caiotpereira**
Split the aktualizr-lite-apps and aktualizr-lite-offline from main aktualizr package.
    
 Aktualizr has been split into two new packages to better track dependencies.
 Now the aklite-offline and aklite-apps can  be provided to custom sota client  without needing to include the aktualizr package.
  
    New Packages:
     - aktualizr-lite-offline:  /usr/bin/aklite-offline
     - aktualizr-lite-apps:    /usr/bin/aklite-apps

### CODE REVIEW

**Review by quaresmajose — APPROVED** [review]
LGTM, with only a minor missing space at the end

File: meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend [hunk 1]
```diff
@@ -54,17 +54,21 @@ do_install:append:lmp() {
...
 FILES:${PN}-lite-dev = "${includedir}/${PN}-lite"
-
+FILES:${PN}-offline = " \
+                    ${bindir}/aklite-offline \
+                    "
 # Force same RDEPENDS, packageconfig rdepends common to both
-RDEPENDS:${PN}-lite = "${RDEPENDS:aktualizr} skopeo"
+RDEPENDS:${PN}-lite = "${RDEPENDS:aktualizr} ${PN}-apps ${@bb.utils.contains('PACKAGECONFIG', 'aklite-offline', 'aktualizr-offline', '', d)} skopeo"
 RDEPENDS:${PN}-lite-lib = "${RDEPENDS:aktualizr} skopeo"
+RDEPENDS:${PN}-apps = "docker-compose skopeo"
```
[the remark this answers is not in the document]
Comment (caiotpereira) [reply meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend:74]: Thanks @quaresmajose

File: meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend [hunk 2]
```diff
@@ -54,17 +54,21 @@ do_install:append:lmp() {
     install -m 0644 ${WORKDIR}/tmpfiles.conf ${D}${nonarch_libdir}/tmpfiles.d/aktualizr-lite.conf
 }
 
-PACKAGES += "${PN}-get ${PN}-lite ${PN}-lite-lib ${PN}-lite-dev"
+PACKAGES += "${PN}-apps ${PN}-get ${PN}-lite ${PN}-lite-lib ${PN}-lite-dev ${PN}-offline "
+FILES:${PN}-apps = " \
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend:58]: Before it was part of aktualizr-lite, and now it is getting moved to aktualizr-apps, while it should be aktualizr-lite-apps, right?

Same for offline.

File: meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start.bb [hunk 3]
```diff
@@ -25,3 +25,4 @@ do_install() {
 }
 
 FILES:${PN} += "${systemd_system_unitdir}/*.service"
+RDEPENDS:${PN} = "aktualizr-lite-apps docker-compose"
```
Comment (quaresmajose) [inline meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start.bb:28]: `docker-compose` is already a runtime dependency of  `aktualizr-lite-apps` so is not need here again.

**Comment by caiotpereira** [comment]
I'm waiting for the https://github.com/foundriesio/aktualizr-lite/pull/250 to be merged to bump the SRCREV:lmp and merge.

File: meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start.bb (hunk 3 above)
Comment (caiotpereira) [reply meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start.bb]: Thank you, fixed .

**Comment by quaresmajose** [comment]
Needs a rebase and maybe can be squashed.

**Comment by ricardosalveti** [comment]
Please squash the changes, try to avoid patches in the series changing things that are added by other patches at the start of the same patch series.

I would do the aktualizr-lite bump first (and adding the new cmake-based option, enabled by default), then have another patch doing the packaging changes.

Also, please update the patches subject from 'meta-lmp-base' to 'base' as a way to reduce the subject length.

File: meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend [hunk 4]
```diff
@@ -80,4 +80,4 @@ RDEPENDS:${PN}-lite = "\
     ${@bb.utils.contains('PACKAGECONFIG', 'aklite-offline', '${PN}-lite-offline', '', d)} \
 "
 RDEPENDS:${PN}-lite-lib = "${RDEPENDS:aktualizr}"
-RDEPENDS:${PN}-lite-apps = "docker-compose skopeo"
\ No newline at end of file
+RDEPENDS:${PN}-lite-apps = "docker-compose skopeo"
```
Comment (mike-sul) [inline meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend:83]: Is this added mistakenly? If not, then maybe should be in dedicated commit.

File: meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start.bb [hunk 5]
```diff
@@ -25,3 +25,4 @@ do_install() {
 }
 
 FILES:${PN} += "${systemd_system_unitdir}/*.service"
+RDEPENDS:${PN} = "aktualizr-lite-apps"
```
Comment (mike-sul) [inline meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start.bb:28]: Is this added mistakenly? If not, then maybe should be in dedicated commit.

Comment (caiotpereira) [reply meta-lmp-base/recipes-support/compose-apps-early-start/compose-apps-early-start.bb]: Removed.

**Comment by quaresmajose** [comment]
The commit message still use `meta-lmp-base` but `base` is what we commonly use and is better to reduce the subject length, the current PR title looks good to be use don the subject.
It would also be good on the commmit message if the number of chars per line was less than about 80 to be easier to read.

**Comment by ricardosalveti** [comment]
@caiotpereira commit message still has meta-base, change to 'base: ...' instead.

Also, what happened with the aktualizr rev bump here?

**Comment by caiotpereira** [comment]
@rsalveti , Fixed the commit message.
The atkualizr rev bump it will be added in the next PR without the new compilation arguments.

File: meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend (hunk 4 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend:83]: Yeah, docker-compose is new here.

Comment (ricardosalveti) [reply meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend:83]: Will merge anyway as this should change the behavior, but better to separate next time.
